### PR TITLE
Refactoring the code using DRY, YAGNI, KISS, and SOLID principles

### PR DESCRIPTION
**Pull request by youmari**

### CODE REVIEW

File: book_ui.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @books = []
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
```
Comment (TedLivist) [inline book_ui.rb]: - [ ] The sole responsibility of this **create_book** method should simply be to create books. The requests for user input and input validation should be handled by another method to obey the single responsibility rule of SOLID. Kindly extract these requests for inputs into another method.

  **SUGGESTION:** You can extract these requests methods into another method and have that method return and array of **title** and **author**. Then destructure the variables in **create_book** like so:

  elshamy-refactor

File: person_ui.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    print 'Name: '
+    @name = gets.chomp
+
+    print 'Has parent_permission ? [Y/N]: '
+    @permission = gets.chomp
+
+    @permission = false if @permission == 'n'
+    @permission = true if @permission == 'y'
+
```
Comment (TedLivist) [inline person_ui.rb]: - [x] Here, the requests for **`age`** and **`name`** break 2 rules: the single-responsibility rule (should be handled by another method) and the DRY rule (it is repeated in the **create_teacher** method). Kindly extract these requests into another method and use that method instead.
- [x] The user inputs and validation for **`@permission`** should be handled by another method. Kindly do that.

  **SUGGESTION:**

  elshamy-person

File: person_ui.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  def create_teacher
+    print 'Age: '
+    @age = gets.chomp
+
+    print 'Name: '
+    @name = gets.chomp
+
+    print 'Specialization: '
+    @specialization = gets.chomp
```
Comment (TedLivist) [inline person_ui.rb]: - [x] Kindly ensure the user inputs are handled by another method instead.

File: rental_ui.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  def create_rental(books, people)
+    get_book_from_user(books)
+    book_num = gets.chomp
+
+    get_person_from_user(people)
+    person_num = gets.chomp
+
+    puts 'Date: '
+    date = gets.chomp
+
```
Comment (TedLivist) [inline rental_ui.rb]: - [x] Kindly ensure user inputs and validation are handled by another method.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @youmari,

Good job so far!
✔️ Good documentation
✔️ Methods are refactored into different classes
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental_ui.rb (hunk 4 above)
Comment (youmari) [reply rental_ui.rb]: thanks for the feedback

**Review by FrederickMih — APPROVED** [review]
# :fire:  Status : :APPROVED :champagne: :champagne: 

Hey @youmari , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**